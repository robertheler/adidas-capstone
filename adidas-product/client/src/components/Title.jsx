import React, { Component, Fragment } from 'react';
import styled from 'styled-components';

const StyledProductName = styled.h1`
  font-size: 52px;
  font-family: AdihausDIN Cn,Helvetica;
  font-weight: 500;
  margin: 0 0 3px 0;
  word-wrap: normal;
  text-transform: uppercase;
  letter-spacing: 1.5px;
  line-height: 1;
`;

const StyledColorName = styled.div`
  font-size: 11px;
  font-family: AdihausDIN,Helvetica;
  font-weight: 400;
  margin: 0 0 30px 0;
  word-wrap: normal;
  text-transform: uppercase;
  letter-spacing: 2px;
  word-wrap: normal;
  color: #767677;
`;

class Collection extends Component {
  constructor(props){
    super(props);
  }

  render() {
    return (
      <Fragment>
        <StyledProductName>{this.props.product.name}</StyledProductName>
        <StyledColorName>{this.props.product.colors[this.props.color].name}</StyledColorName>
      </Fragment>
    );
  }
}

export default Collection;