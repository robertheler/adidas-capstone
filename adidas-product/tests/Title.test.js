import React from 'react';
import { shallow, mount, render } from 'enzyme';
import Title from '../client/src/components/Title';

// json data will be used to generate props
var data = require('../db/data.json');

function setup() {
  const props = {
    color: 0,
    product: data[0]
  };
  const wrapper = shallow(<Title color={props.color} product={props.product} isSelected={props.isSelected} handleColorChange={props.handleColorChange}/>);
  return { wrapper, props };
}

describe('Title Component Test Suite', () => {
  const { wrapper, props } = setup();

  it('It should render', () => {
    expect(wrapper).toHaveLength(1);
  });

  it('It should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});