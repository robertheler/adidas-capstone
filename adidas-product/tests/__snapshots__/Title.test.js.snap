// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title Component Test Suite It should render correctly 1`] = `
<Fragment>
  <styled.h1>
    NMD_R1 Shoes
  </styled.h1>
  <styled.div>
    Black / Black / Gum
  </styled.div>
</Fragment>
`;
